cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -222,114 +218,30 @@ func (he *hcsExec) Start(ctx context.Context) (err error) {
 			}
 		}()
 	}
-	var (
-		proc cow.Process
-	)
-	if he.isWCOW {
-		wpp := &hcsschema.ProcessParameters{
-			CommandLine:      he.spec.CommandLine,
-			User:             he.spec.User.Username,
-			WorkingDirectory: he.spec.Cwd,
-			EmulateConsole:   he.spec.Terminal,
-			CreateStdInPipe:  he.io.StdinPath() != "",
-			CreateStdOutPipe: he.io.StdoutPath() != "",
-			CreateStdErrPipe: he.io.StderrPath() != "",
-		}
-
-		if he.spec.CommandLine == "" {
-			wpp.CommandLine = escapeArgs(he.spec.Args)
-		}
-
-		environment := make(map[string]string)
-		for _, v := range he.spec.Env {
-			s := strings.SplitN(v, "=", 2)
-			if len(s) == 2 && len(s[1]) > 0 {
-				environment[s[0]] = s[1]
-			}
-		}
-		wpp.Environment = environment
-
-		if he.spec.ConsoleSize != nil {
-			wpp.ConsoleSize = []int32{
-				int32(he.spec.ConsoleSize.Height),
-				int32(he.spec.ConsoleSize.Width),
-			}
-		}
-		proc, err = he.c.CreateProcess(wpp)
-	} else {
-		lpp := &lcow.ProcessParameters{
-			ProcessParameters: hcsschema.ProcessParameters{
-				CreateStdInPipe:  he.io.StdinPath() != "",
-				CreateStdOutPipe: he.io.StdoutPath() != "",
-				CreateStdErrPipe: he.io.StderrPath() != "",
-			},
-		}
-		if he.id != he.tid {
-			// An init exec passes the process as part of the config. We only pass
-			// the spec if this is a true exec.
-			lpp.OCIProcess = he.spec
-		}
-		proc, err = he.c.CreateProcess(lpp)
+	cmd := &hcsoci.Cmd{
+		Host:   he.c,
+		Stdin:  he.io.Stdin(),
+		Stdout: he.io.Stdout(),
+		Stderr: he.io.Stderr(),
+		Log: logrus.WithFields(logrus.Fields{
+			"tid": he.tid,
+			"eid": he.id,
+		}),
+		CopyAfterExitTimeout: time.Second * 1,
+	}
+	if he.isWCOW || he.id != he.tid {
+		// An init exec passes the process as part of the config. We only pass
```

Comment:
This is confusing now that you have combined them. We pass spec for WCOW in all cases. For LCOW the init process spec was passed at create. So we just call start. Can we rewrite this to make it more clear?

Reply:
Could have but you merged it.

Reply:
It was 5:50 in the morning. My bad